src/resource_filter.ts

```diff
@@ -0,0 +1,179 @@
+import { getProperty, hasProperty } from "dot-prop";
+import mongoose from "mongoose";
+import { detective } from "./detective";
+import { QueryOptions } from "./getQuerySchema";
+import { Access, ExcludedMap, FilteredKeys } from "./types";
+import { weedout } from "./weedout";
+
+export class Filter {
+  excludedMap: ExcludedMap = new Map();
+
+  add(
```

Comment:
Rather than intanciating one filter per schema, there is not a single filter and we add schemas to it. This allowed me to encapsulate `excludedMap` and not have to carry it around and not to rely on mutations and side-effects.